configs/components/curl.rb

```diff
@@ -33,6 +33,9 @@
 
   configure_options = []
   configure_options << "--with-ssl=#{settings[:prefix]}"
+  if !settings[:use_legacy_openssl_algos]
```

Comment:
I'm thinking we don't want to affect curl in these runtime projects:

* agent-runtime-7.x
* agent-runtime-6.x (this will be deleted soon)
* client-tools-runtime-2021.7.x
* client-tools-runtime-2019.8.x (this will be deleted soon)

But we do want to for:

* agent-runtime-main

I'm not sure about:

* pdk
* pe-installer-runtime-main
* pe-installer-runtime-2021.7.x
* pe-installer-runtime-2019.8.x (this will be deleted soon)
* pe-bolt-server-runtime-main
* pe-bolt-server-runtime-2021.7.x
* pe-bolt-server-runtime-2019.8.x (to be deleted soon)

Reply:
I just pushed an update that would conditionally disable ntlm only for openssl version 3 projects, which I think is just agent-runtime-main and pdk. This change was part of my heads up to the pdk team so I _think_ that's okay. Or at least they'll get back to me. I'm testing the change locally, let me know what you think?